### PR TITLE
refactor: Switch to `wagmi`.

### PR DESCRIPTION
**Pull request by mprasanjith**
# Description

This PR removes `web3-react` from the DAVI codebase in favor of much newer and better maintained `wagmi`.

Closes https://github.com/DXgovernance/DAVI/issues/69.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Existing test suite has been updated to work with `wagmi`. Most of DAVI has been tested manually, but additional testing is welcome!

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by rossneilson** [comment]
With my wallet connected if I change network I get the wallet prompt and it changes but the app's network is not changing with it and it becomes impossible to change without a refresh

**Comment by rossneilson** [comment]
Also seeing behaviour where in frame if I switch network manually in the wallet the dapp's network doesn't update but if I go to change the dapp in the network modal it doesn't do anything

File: src/hooks/Guilds/web3/useJsonRpcProvider.ts [hunk 1]
```diff
@@ -1,17 +1,6 @@
...
-
-  const { library, chainId: walletChainId } = useWeb3React();
-  return useMemo(
-    () =>
-      (!chainId || walletChainId === chainId) && library
-        ? new providers.Web3Provider(library.currentProvider)
-        : multichainProviders[chainId],
-    [library, chainId, walletChainId, multichainProviders]
-  );
+export default function useJsonRpcProvider() {
```
Comment (MiltonTulli) [inline src/hooks/Guilds/web3/useJsonRpcProvider.ts:3]: Shouldn't we remove the entire `useJsonRpcProvider` and use `useProvider` across the app?

**Review by MiltonTulli — APPROVED** [review]
Didn't test it but the code looks good to me. Amazing refactor.
Left minor coments.

**Comment by dcrescimbeni** [comment]
Very cool! Everything seems OK.

Waiting for the synpress fix to approve.

File: cypress/support/pageObjects/Guilds.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ class Guilds {
   goToGuildsPage(address?: string) {
     const baseUrl = Cypress.config('baseUrl');
     const network = Cypress.env('network');
-    cy.visit(`${baseUrl}/guilds/${network}${address ? `/${address}` : ''}`, {
```
[the remark this answers is not in the document]
Comment (mprasanjith) [reply cypress/support/pageObjects/Guilds.ts:50]: Discussed with Ross yesterday and confirmed we're going to.

**Comment by rossneilson** [comment]
I am still seeing white screen issues when I do not have local hardhat running

**Review by MiltonTulli — DISMISSED** [review]
Great work!

**Comment by rossneilson** [comment]
The behaviour seems good to me but still seeing white screen issues when not running localhost

**Review by rossneilson — APPROVED** [review]

**Pull request merged**